app/src/main/java/com/hedvig/app/feature/offer/ui/OfferAdapter.kt

```diff
@@ -418,4 +498,27 @@ class OfferAdapter(
             }
         }
     }
+
+    companion object {
+        const val INSURELY_CARD = 1
+        const val CURRENT_INSURANCES_HEADER = 2
+        const val INSURELY_DIVIDER = 3
+
+        class OfferDiffUtilCallback : DiffUtil.ItemCallback<OfferModel>() {
+            override fun areItemsTheSame(oldItem: OfferModel, newItem: OfferModel): Boolean = when {
+                oldItem is OfferModel.InsurelyCard && newItem is OfferModel.InsurelyCard -> {
+                    oldItem.id == newItem.id
+                }
+                oldItem is OfferModel.CurrentInsurancesHeader && newItem is OfferModel.CurrentInsurancesHeader -> {
+                    // Should only display 1 CurrentInsurancesHeader ever
+                    true
+                }
+                else -> {
+                    oldItem == newItem
```

Comment:
Can we check id here?

Reply:
Not all of them (if any) have a proper ID, so just carried over the exact same functionality as we had before (GenericDiffUtilItemCallback) which simply checks for object equality. Putting an ID to all the items is a bit out of scope for this and also maybe not needed since we're not really using the RV in this screen to update any of those items on the fly